### ISSUE TITLE
Feature: Combine multiple runs in parrot_package_create

### ISSUE THREAD

**Issue by dthain** [cooperative-computing-lab/cctools#1405]
Allow the user to add the contents of another run to an existing parrot package, detecting conflicts if necessary.  For example:

```
parrot_package_run   somecommand
parrot_package_create mypackage
parrot_package_run   anothercommand
parrot_package_create  --add mypackage
```

**Comment by dthain** [comment]
The trick here is that if the user ask to add someting that conflicts with the original (e.g. same file but different contents), give them some useful options.

**Comment by hmeng-19** [comment]
@dthain, just to ensure we are on the same page. `parrot_package_run` is used
to run tasks inside a package created by `parrot_package_create`, and the
inputs of `parrot_package_create` are the envlist and namelist created by
`parrot_run --envlist envlist.1 --namelist namelist.1 command`.
 
I believe the following workflow is what you meant (Please correct me if you do not agree):

    parrot_run --env-list envlist1 --name-list namelist1 command1
    parrot_package_create --env-list envlist1 --name-list namelist1 --package-path mypackage1
    parrot_package_run --env-list envlist2 --name-list namelist2 command2
 
    // this command will add the new files recorded in namelist2 into mypackage1.
    parrot_package_create  --env-list envlist2 --name-list namelist2 --add mypackage1
 
My only concern is whether we should merge envlist1 and envlist2?

If we merge them, then command1 may become unreproducible due to the broken environment variable settings.
 
My suggestion is that we put both `envlist1` and `envlist2` into `mypackage1`,
and add a new option into `parrot_pacakge_run`, `--env-list`.

If the user want to reproduce command1, he should run:

    parrot_package_run --env-list mypackage1/envlist1 --package-path mypackage1
 
If the user want to reproduce command2, he should run:

    parrot_package_run --env-list mypackage1/envlist2 --package-path mypackage1

**Comment by hmeng-19** [comment]
@dthain , I created a PR #1406 to fix this, please take a look.

**Comment by dthain** [comment]
If I understand correctly, you are allowing the user to choose on of multiple environment lists at runtime.
I think what Bill was requesting was the ability to add to the originally stored environment list, not create a separate one.

e.g. You run `parrot_package_create` once, then realize that you need to add one more file and one more environment variable, so you want a command that lets you modify the existing list, not create a new one.

**Comment by hmeng-19** [comment]
@dthain, I did not think so.

As far as I understand, Bill really wants to add new files into an existing
package, which is solved by the new feature:

    parrot_package_create --name-list newnamelist --add existingpackage

As for the environment file, you really do not want to modify the origin
environment file in place. That may make the original application completely
non-reproducible.

**Comment by hmeng-19** [comment]
@dthain , one thing we must realize is that the package created for one application can not be directly used to run another application without mending the package first.
 
    parrot_run --env-list envlist1 --name-list namelist1 uname
    parrot_package_create --env-list envlist1 --name-list namelist1 --package mypack
    parrot_package_run --package mypack uname
 
Here, using `mypack` you can rerun `uname` command. However, you can not run `date` command, because the depdencies of `date` is not inside `mypack`.

**Comment by billobob** [comment]
As far as environments go, it's probably better to keep them separate for now. 

Another issue I've noticed is when running packages, there's often a problem due to the way parrot mounts files (using the GID/UID of whoever executes parrot_package_run)--this can cause some weird stuff to happen when a program is re-executed on different machines. As a temporary solution (and maybe permanent one) I've just copied the /etc/passwd and /etc/group from the and created the current users home directory before running a package. [Note that running as root of course works, but i'm not assuming end users will have root/privileged access]

It's not an issue for me, but I just thought it's something you guys should be aware of.
